docs/DeveloperGuide.md

```diff
@@ -312,32 +312,196 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `TalentHub` and the **Actor** is the `celebrity talent manager`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - Add Contact**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. Talent Manager requests to add a specific contact
+2. TalentHub adds the person
 
-    Use case ends.
+Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. The command format is incorrect.
 
-  Use case ends.
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
 
-* 3a. The given index is invalid.
+    Use case resumes at step 1.
+  
+* 1b. Any parameter is missing or invalid.
 
-    * 3a1. AddressBook shows an error message.
+  * 1b1. TalentHub outputs an error message specifying the issue.
 
-      Use case resumes at step 2.
+    Use case resumes at step 1.
+
+* 1c. An identical phone number is detected.
+
+  * 1c1. TalentHub outputs an error message specifying the issue.
+
+    Use case resumes at step 1.
+
+**Use case: UC02 - Delete Contact after List**
+
+**MSS**
+
+1. Talent Manager [list contacts (UC06)](#)
+2. Talent Manager [delete contact (UC04)](#)
+
+   Use case ends.
+
+**Use case: UC03 - Delete Contact after List**
+
+**MSS**
+
+1. Talent Manager [find contacts (UC05)](#)
+2. Talent Manager [delete contact (UC04)](#)
+
+   Use case ends.
+
+**Use case: UC04 - Delete Contact**
+
+**MSS**
+
+1. Talent Manager requests to delete a specific person in the list
+2. TalentHub deletes the person
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command format is incorrect.
+
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
+
+    Use case resumes at step 1.
+
+* 1b. The list is empty.
+
+  * 1b1. System shows an error message specifying the list is empty.
+
+    Use case resumes at step 1.
+
+* 1c. The given index is invalid.
+
+  * 1c1. System shows an error message.
+
+    Use case resumes at step 1.
+
+**Use case: UC05 - Find Contact**
+
+**MSS**
+
+1. User requests to find persons with with `keywords`
+2. System processes and list person with `keywords`
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command format is incorrect.
+
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
+
+    Use case resumes at step 1.
+
+* 1b. The keyword is empty.
+
+  * 1b1. System shows an error message.
+
+    Use case resumes at step 1. 
+
+
+**Use case: UC06 - List Contact**
+
+**MSS**
+
+1. Talent Manager requests to list persons
+2. System shows a list of persons
```

Comment:
Same for these portions, specify we're listing all

Reply:
Fixed